### PR TITLE
Forward unittest workloads to AWS

### PR DESCRIPTION
**Pull request by yunchu**
### Summary

This PR is for moving unit test jobs from the dedicated internal resource to the runner on the AWS.

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Comment by eunwoosh** [comment]
It seems that unit test result isn't passed to Github. Is it ok?

**Review by goodsong81 — DISMISSED** [review]

**Comment by yunchu** [comment]
> It seems that unit test result isn't passed to Github. Is it ok?

oh.. that's my fault. I will push the patch to enable coverage reporting. thanks.

**Review by goodsong81 — APPROVED** [review]

File: .github/workflows/pre_merge.yml [hunk 1]
```diff
@@ -34,15 +34,11 @@ jobs:
...
         run: tox -vv -e pre-commit-all-py310-pt1
   Unit-Test:
-    runs-on: [self-hosted, linux, x64, dev]
     needs: Code-Quality-Checks
-    timeout-minutes: 120
     strategy:
       fail-fast: false
       matrix:
         include:
-          - python-version: "3.8"
```
Comment (eunwoosh) [inline .github/workflows/pre_merge.yml:44]: Don't we need to run unit test w/ python 3.8?

**Review by eunwoosh — COMMENTED** [review]
image

This PR is waiting for unit test status as above.
Will it be fixed in the develop branch after it's merged?

File: .github/workflows/pre_merge.yml [hunk 2]
```diff
@@ -52,19 +48,37 @@ jobs:
...
+      image: 219678651685.dkr.ecr.eu-central-1.amazonaws.com/ote-ci:pr-194-1334bef936df3b06d28bc3d1bd90e7248359b99e
+      options: --runtime=nvidia --env-file=/home/runner/.nvidia.env --shm-size=24g
     steps:
       - name: Checkout repository
         uses: actions/checkout@v3
       - name: Install Python
         uses: actions/setup-python@v4
         with:
-          python-version: ${{ matrix.python-version }}
+          python-version: "3.8"
```
Comment (chuneuny-emily) [inline .github/workflows/pre_merge.yml:76]: Didn't you delete Python 3.8?

File: .github/workflows/pre_merge.yml (hunk 1 above)
Comment (yunchu) [reply .github/workflows/pre_merge.yml:44]: unit tests will be ran with python 3.9 and 3.10 and the unit tests with coverage will be ran for python 3.8.

File: .github/workflows/pre_merge.yml (hunk 2 above)
Comment (yunchu) [reply .github/workflows/pre_merge.yml:76]: this is just for the coverage run.

**Comment by yunchu** [comment]
> image
> This PR is waiting for unit test status as above. Will it be fixed in the develop branch after it's merged?

required list of check should be changed due to the modification of pr-merge workflow. once this change is merged, i will update the list.

**Review by eunwoosh — APPROVED** [review]
Thanks for your work :)

File: .github/workflows/pre_merge.yml (hunk 2 above)
Comment (chuneuny-emily) [reply .github/workflows/pre_merge.yml:76]: Python 3.8 was put into EOL last year, shouldn't this be made to 3.10 as well?

**Pull request merged**